### PR TITLE
[fix][issues-11517] cannot select project when create depend task

### LINKED ISSUE #11517: [Bug] [Workflow Definition] ordinary users can not create depend task
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image

### What you expected to happen

create project, but can not select  it

### How to reproduce

ordinary users  create poject

### Anything else

_No response_

### Version

3.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rusonding**
## Brief change log

close #11517 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml [hunk 1]
```diff
@@ -167,8 +167,8 @@
         </include>
         from t_ds_project p
         <if test="userId != 0">
-            right join t_ds_relation_project_user rpu on p.id = rpu.project_id
-            where rpu.user_id = #{userId}
+            left join t_ds_relation_project_user rpu on p.id = rpu.project_id
+            where rpu.user_id = #{userId} or p.user_id=#{userId}
```
Comment (SbloodyS) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml:171]: I think ordinary user should not be allowed to query its project before it's authorized.

**Comment by caishunfeng** [comment]
Hi @rusonding , as @SbloodyS said, the ordinary user should not be allowed to query the project without authorize.

I will close this pr, you can reopen it if you want it.

**Comment by caishunfeng** [comment]
image

I think it's better to auto authorize the project to its creater, WDYT?

**Comment by caishunfeng** [comment]
close by #11961

**Pull request closed without merging**